### ISSUE TITLE
The header settings doesn't work after adding i18n settingss to nextconfig for the files under pages folder

### ISSUE THREAD

**Issue by LIU-CE-01** [vercel/next.js#58291]
### Link to the code that reproduces this issue

https://github.com/LIU-CE-01/nextjs-bug-report

### To Reproduce

I'm using nextjs 13.5.6 and I added some header settings and i18n settings in the next.config.js

1. npm run dev
2. Go to /about page (under pages folder) and check the response header
3. Commont i18n settings in the next.config.js
4. npm run dev again and Go to /about page and check the response header

### Current vs. Expected behavior

Current:
When i18n settings is included, csp is not included in the response headers.
When i18n settings is commented, csp is included in the response headers.

Expected:
csp should be included in the response headers no matter whether we have i18n settings in next.config.js

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: MacOS Sonoma
Binaries:
  Node: 18.17.1
  npm: 9.6.7
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 13.5.6,
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Internationalization (i18n)

### Additional context

We have used the version 13.4.3 and it seems we don't have the same issue

**Comment by cosieLq** [comment]
Related to #54159

**Comment by cosieLq** [comment]
Eventually we had to write a middleware to solve this headers problem.

i18n settings seem to be rather instable for long, since somewhere around next v13.4. The routing server doesn't always work correctly. With some versions of next, the files in the public folder return 404. We also had to use middleware to make it work.

In the docs for App Router, they don't even mention i18n in next.config. I assume it's not supported any more, but it should then be communicated explicitly right?

**Comment by LIU-CE-01** [comment]
@cosieLq  Yeah, they need to make it clear whether it is supported or not.

Thank you for your advice, we will try to use middleware to solve the problem and waiting for their reponse

**Comment by ztanner** [comment]
Hi, I wasn't able to reproduce the behavior you described with your reproduction on the latest canary. Response headers from the `/about` route appear to correctly have your CSP headers.

![CleanShot 2023-12-04 at 15 15 46@2x](https://github.com/vercel/next.js/assets/1939140/6bef7733-15ec-4840-8484-b388867ad399)

If the concern is that the `/_next/static` resources don't contain these CSP headers, please refer to [this](https://nextjs.org/docs/app/api-reference/next-config-js/headers#headers-with-i18n-support) docs page which indicates how you can use the `locale: false` setting to ensure all routes are matched regardless of i18n. 

Please don't hesitate to ping or open another issue if you are still encountering problems!

**Comment by LIU-CE-01** [comment]
@ztanner 
Thanks for your reply. I tried the new canary for nextjs 14 and it seems the issue is fixed.

But currently our team is prefer to stay in nextjs 13. I tried the newest canary for 13.5.7 (13.5.7-canary.37) and I can still reproduce the issue.
<img width="597" alt="Screenshot 2023-12-05 at 9 34 56" src="https://github.com/vercel/next.js/assets/111339665/6fb90ee7-7ee6-4dfc-8a92-a66f93eb2984">

As there is still a canary, we believe nextjs 13 is still supported. Is there any plan to fix it in nextjs 13?
(I updated the package json in the branch to point to the version 13.5.7-canary.37)
